inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -439,4 +718,107 @@ InferenceEngine::Precision MKLDNNDeconvolutionNode::getRuntimePrecision() const
+    m_memFrom = std::make_shared<MKLDNNMemory>(engine);
+    m_memFrom->Create(MKLDNNExtensionUtils::makeDescriptor(descFrom));
+    m_reorder = mkldnn::reorder(m_memFrom->GetPrimitive(), m_memTo->GetPrimitive());
+}
+
+void MKLDNNDeconvolutionNode::DeconvExecutor::IntermReorder::exec(mkldnn::stream strm) {
+    auto src = m_memFrom->GetPrimitive();
+    auto dst = m_memTo->GetPrimitive();
+    m_reorder.execute(strm, src, dst);
+}
+
+void MKLDNNDeconvolutionNode::DeconvExecutor::exec(mkldnn::stream strm) {
+    for (auto &inReorder : inputReorders) {
+        inReorder.exec(strm);
+    }
+    (*execPrim).execute(strm, primArgs);
+    for (auto &outReorder : outputReorders) {
+        outReorder.exec(strm);
+    }
+}
+
+MKLDNNDeconvolutionNode::DeconvExecutorDefault::DeconvExecutorDefault(const mkldnn::convolution_backward_data::primitive_desc& pd,
+                                                                      MKLDNNMemoryPtr inMem,
+                                                                      MKLDNNMemoryPtr weightMem,
+                                                                      MKLDNNMemoryPtr outMem,
+                                                                      const mkldnn::engine& engine) {
+    execPrim.reset(new mkldnn::convolution_backward_data(pd));
+
+    if (inMem->GetPrimitive().get_desc() != pd.diff_dst_desc()) {
+        inputReorders.push_back(IntermReorder(inMem, pd.diff_dst_desc(), engine));
+        primArgs[DNNL_ARG_DIFF_DST] = inputReorders.back().getToMem()->GetPrimitive();
+    } else {
+        primArgs[DNNL_ARG_DIFF_DST] = inMem->GetPrimitive();
+    }
+
+    if (weightMem->GetPrimitive().get_desc() != pd.weights_desc()) {
+        inputReorders.push_back(IntermReorder(weightMem, pd.weights_desc(), engine));
+        primArgs[DNNL_ARG_WEIGHTS] = inputReorders.back().getToMem()->GetPrimitive();
+    } else {
+        primArgs[DNNL_ARG_WEIGHTS] = weightMem->GetPrimitive();
+    }
+
+    if (outMem->GetPrimitive().get_desc() != pd.diff_src_desc()) {
+        outputReorders.push_back(IntermReorder(pd.diff_src_desc(), outMem, engine));
+        primArgs[DNNL_ARG_DIFF_SRC] = outputReorders.back().getFromMem()->GetPrimitive();
+    } else {
+        primArgs[DNNL_ARG_DIFF_SRC] = outMem->GetPrimitive();
+    }
+}
+
+MKLDNNDeconvolutionNode::DeconvExecutorInt8::DeconvExecutorInt8(const mkldnn::deconvolution_forward::primitive_desc& pd,
+                                                                MKLDNNMemoryPtr inMem,
+                                                                MKLDNNMemoryPtr weightMem,
+                                                                MKLDNNMemoryPtr outMem,
+                                                                const mkldnn::engine& engine) {
+    execPrim.reset(new mkldnn::deconvolution_forward(pd));
+
+    if (inMem->GetPrimitive().get_desc() != pd.src_desc()) {
```

Comment:
Lines 789-794: extract as separate method into base class?

Reply:
I plan to extract `DeconvExecutor` in common part and use it for other oneDnn primitive. See: https://github.com/openvinotoolkit/openvino/pull/8871#discussion_r763955139 Also first deconv input and output are different for INT8 and FP32 deconv.

Reply:
Sounds good!